velox/functions/lib/string/StringImpl.h

```diff
@@ -199,27 +199,52 @@ std::vector<int32_t> stringToCodePoints(const T& inputString) {
 
 /// Returns the starting position in characters of the Nth instance(counting
 /// from the left if lpos==true and from the end otherwise) of the substring in
-/// string. Positions start with 1. If not found, 0 is returned. If subString is
-/// empty result is 1.
+/// string after position 'start'. Positions start with 1. If not
+/// found, 0 is returned. If subString is empty result is 1.
+/// @param instance The instance of the substring to find in string.
+/// @param start The start position in characters to search for the substring in
+/// string.
 template <bool isAscii, bool lpos = true, typename T>
-FOLLY_ALWAYS_INLINE int64_t
-stringPosition(const T& string, const T& subString, int64_t instance = 0) {
+FOLLY_ALWAYS_INLINE int64_t stringPosition(
+    const T& string,
+    const T& subString,
+    int64_t instance = 0,
+    int64_t start = 1) {
   VELOX_USER_CHECK_GT(instance, 0, "'instance' must be a positive number");
   if (subString.size() == 0) {
     return 1;
   }
+  if (start < 1 || start > string.size()) {
+    return 0;
+  }
+
+  int64_t startByteIndex;
+  if constexpr (isAscii) {
+    startByteIndex = start - 1;
+  } else {
+    // Calculate the byte index of the start character.
+    const char* pos = string.data();
+    int64_t numCharacters = 0;
+    while (numCharacters < start - 1) {
+      if (!utf_cont(*pos)) {
+        numCharacters++;
+      }
+      pos++;
+    }
+    startByteIndex = pos - string.data();
+  }
+
+  // The string to search for substring.
+  auto view = std::string_view(
```

Comment:
maybe we can pass the `startByteIndex` to `findNthInstanceByteIndexFromStart`, instead of create a new string_view object.

Reply:
`findNthInstanceByteIndexFromEnd` does not accept the `startByteIndex` parameter, and we also use `view` at L255.

Reply:
Ok, `stringPosition` also supports to serach from end.

Reply:
But it's little wired, this function searches from end but with `start` parameter as value 1.

Reply:
I understand your point. Updated this PR to make the start the end position when searching from the end. Thanks.